presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2019,15 +2024,43 @@ private PhysicalOperation createLookupJoin(JoinNode node,
                     spillEnabled && !buildOuter && partitionCount > 1,
                     singleStreamSpillerFactory);
 
-            context.addDriverFactory(
-                    buildContext.isInputDriver(),
-                    false,
-                    ImmutableList.<OperatorFactory>builder()
-                            .addAll(buildSource.getOperatorFactories())
-                            .add(hashBuilderOperatorFactory)
-                            .build(),
-                    buildContext.getDriverInstanceCount(),
-                    buildSource.getPipelineExecutionStrategy());
+            if (SystemSessionProperties.isEnableDynamicFiltering(context.getSession()) && node.getType() == INNER && node.getCriteria().size() > 0) {
+                List<Symbol> filterSymbols = node.getDynamicFilters().values().stream().collect(toImmutableList());
+                List<String> dynamicFilterSymbolNames = node.getDynamicFilters().keySet().stream().collect(toImmutableList());
```

Comment:
nit: can we use `ImmutableList.copyOf(...)` also for `filterSymbols` and `dynamicFilterSymbolNames`?

Reply:
Done